src/demo/js/script.js

```diff
@@ -156,6 +156,29 @@ let preview = {
     this.update();
     return false;
   },
+  reset: function () {
+    const inputs = document.querySelectorAll(".param");
+    inputs.forEach((input) => {
+      let value = this.defaults[input.name];
+      if (value) {
+        input.value = value;
+      }
+    });
+    const overrides = {
+      font: "Fira Code",
+      pause: "1000",
+      width: "435",
+    };
+    let value = overrides[input.name] || this.defaults[input.name];
+    input.jscolor.fromString(value);
+    if (value) {
+      if (["color", "background"].includes(input.name)) {
+        input.jscolor.fromString(value);
+      } else {
+        input.value = value;
+      }
+    }
+  }
```

Comment:
This is pretty close, but it seems it is slightly out of order. Retrieving the value from the overrides/defaults, the conditions, and the assignment should all happen within the forEach loop.

See this suggestion below.

```suggestion
  reset: function () {
    const overrides = {
      font: "Fira Code",
      pause: "1000",
      width: "435",
    };
    // reset all inputs
    const inputs = document.querySelectorAll(".param");
    inputs.forEach((input) => {
      let value = overrides[input.name] || this.defaults[input.name];
      if (value) {
        if (["color", "background"].includes(input.name)) {
          input.jscolor.fromString(value);
        } else {
          input.value = value;
        }
      }
    });
  },
```